integration_tests/utils.go

```diff
@@ -88,42 +95,224 @@ func BuildXionChain(t *testing.T) (*cosmos.CosmosChain, context.Context) {
+	initialXionSupply, err := dyno.GetString(xionCoin, "amount")
+	require.NoError(t, err)
+	return initialXionSupply
+}
+
+func getAddressBankBalanceAtHeight(t *testing.T, xion *cosmos.CosmosChain, ctx context.Context, address string, denom string, blockHeight uint64) string {
+	if blockHeight == 0 {
+		blockHeight, _ = xion.Height(ctx)
+		require.Greater(t, blockHeight, 0)
+	}
+	jsonRes := make(map[string]interface{})
+	queryRes, _, err := xion.FullNodes[0].ExecQuery(ctx, "bank", "balances", address, "--height", strconv.FormatInt(int64(blockHeight), 10))
+	require.NoError(t, err)
+
+	require.NoError(t, json.Unmarshal(queryRes, &jsonRes))
+
+	balances, err := dyno.GetSlice(jsonRes, "balances")
+	require.NoError(t, err)
+	if len(balances) == 0 {
+		return "0"
+	}
+	// Make sure we selected the uxion denom
+	balanceDenom, err := dyno.GetString(balances[0], "denom")
+	require.NoError(t, err)
+	require.Equal(t, balanceDenom, denom)
+	balance, err := dyno.GetString(balances[0], "amount")
+	require.NoError(t, err)
+	t.Logf("Balance for address %s: %s", address, balance)
+	return balance
+}
+
+func GetModuleAddress(t *testing.T, xion *cosmos.CosmosChain, ctx context.Context, moduleName string) string {
+	jsonRes := make(map[string]interface{})
+	queryRes, _, err := xion.FullNodes[0].ExecQuery(ctx, "auth", "module-account", moduleName)
+	require.NoError(t, err)
+
+	require.NoError(t, json.Unmarshal(queryRes, &jsonRes))
+
+	moduleAddress, err := dyno.GetString(jsonRes, "account", "base_account", "address")
+	require.NoError(t, err)
+	t.Logf("%s module address: %s", moduleName, moduleAddress)
+	return moduleAddress
+}
+
+// Retrieve a block annual provision
+func GetBlockAnnualProvision(t *testing.T, xion *cosmos.CosmosChain, ctx context.Context, denom string, blockHeight uint64) math.LegacyDec {
+	if blockHeight == 0 {
+		blockHeight, _ = xion.Height(ctx)
+		require.Greater(t, blockHeight, 0)
+	}
+
+	// Query the current block provision
+	var annualProvision json.Number
+	queryRes, _, err := xion.FullNodes[0].ExecQuery(ctx, "mint", "annual-provisions", "--height", strconv.FormatInt(int64(blockHeight), 10))
+	require.NoError(t, err)
+	require.NoError(t, json.Unmarshal(queryRes, &annualProvision))
+	// Query the block per year
+	var params = make(map[string]interface{})
+	queryRes, _, err = xion.FullNodes[0].ExecQuery(ctx, "mint", "params")
+	require.NoError(t, err)
+	require.NoError(t, json.Unmarshal(queryRes, &params))
+	blocksPerYear, err := dyno.GetInteger(params, "blocks_per_year")
+	require.NoError(t, err)
+	// Calculate the block provision
+	return math.LegacyMustNewDecFromStr(annualProvision.String()).QuoInt(math.NewInt(blocksPerYear)) // This ideally is the minted tokens for the block
+}
+
+// This test confirms the property of the module described at
+// https://www.notion.so/burntlabs/Mint-Module-Blog-Post-78f59fb108c04e9ea5fa826dda30a340
+func MintModuleTestHarness(t *testing.T, xion *cosmos.CosmosChain, ctx context.Context, blockHeight int) {
```

Comment:
The mint harness test logic is as follows
Given a particular block height we compute the minted or burned tokens at that block using the following algorithm

- Get the total token supply at previous height 
- Get the total token supply at the current height
- Compute the change in the total supply (d)
- Get the balance of the distribution module account in the previous block
- Get the balance of the distribution module account at the current block
- The difference (f) is the sum of the total fees and minted token at the current block
- The difference ( d - f )is the total fees accrued at the current block
- Get the block provision at the current height (this is the number of tokens to be minted for validators and delegtors at that block
- Perform the following checks 
- If we have more fees accrued than the block provision, we burn the excess tokens 281 - 286
- If instead there are fewer fees than block provision, we only mint (provision - fees) tokens to meet expectations 276 - 281
- Otherwise, we had equal fees and provisions. We do not mint or burn any tokens 286 - 290